Fix set_grad_enabled HOP in strict mode with new tracer

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #163137
* #163136
* #163107
* #162993
* #162992
* #162682
* __->__ #162559
* #162558
* #162557

previous graph seems wrong probably because dynamo bytecode running might be changing the grad state unintentionally.

Differential Revision: [D82478643](https://our.internmc.facebook.com/intern/diff/D82478643)

tugsbayasgalan committed:
Fix set_grad_enabled HOP in strict mode with new tracer

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix set_grad_enabled HOP in strict mode with new tracer"




[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix set_grad_enabled HOP in strict mode with new tracer"




[ghstack-poisoned]

ydwu4 commented on test/export/test_passes.py:
```diff
@@ -835,13 +837,14 @@ def forward(self, x):
-    getitem = add_1[0];  add_1 = None
-    sub = torch.ops.aten.sub.Tensor(getitem, 1)
-    return pytree.tree_unflatten((getitem, sub), self._out_spec)
+    submod_4 = self.submod_1
```
Why it doesn't error out?

tugsbayasgalan committed:
Update on "Fix set_grad_enabled HOP in strict mode with new tracer"


previous graph seems wrong probably because dynamo bytecode running might be changing the grad state unintentionally. 



[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix set_grad_enabled HOP in strict mode with new tracer"


previous graph seems wrong probably because dynamo bytecode running might be changing the grad state unintentionally. 



[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix set_grad_enabled HOP in strict mode with new tracer"


previous graph seems wrong probably because dynamo bytecode running might be changing the grad state unintentionally. 

Differential Revision: D82180554



[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix set_grad_enabled HOP in strict mode with new tracer"


previous graph seems wrong probably because dynamo bytecode running might be changing the grad state unintentionally. 

Differential Revision: D82180554



[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82478643).

tugsbayasgalan committed:
Update on "Fix set_grad_enabled HOP in strict mode with new tracer"


previous graph seems wrong probably because dynamo bytecode running might be changing the grad state unintentionally.

Differential Revision: [D82478643](https://our.internmc.facebook.com/intern/diff/D82478643)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82478643).

tugsbayasgalan committed:
Update on "Fix set_grad_enabled HOP in strict mode with new tracer"


previous graph seems wrong probably because dynamo bytecode running might be changing the grad state unintentionally.

Differential Revision: [D82478643](https://our.internmc.facebook.com/intern/diff/D82478643)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Fix set_grad_enabled HOP in strict mode with new tracer"


previous graph seems wrong probably because dynamo bytecode running might be changing the grad state unintentionally.

Differential Revision: [D82478643](https://our.internmc.facebook.com/intern/diff/D82478643)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82478643).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D82478643).
